src/main/java/dev/jbang/Main.java

```diff
@@ -12,6 +12,7 @@ public static void main(String... args) {
 		CommandLine cli = Jbang.getCommandLine();
 		args = handleDefaultRun(cli.getCommandSpec(), args);
 		int exitcode = cli.execute(args);
+		((Jbang) cli.getCommandSpec().userObject()).checkLatestVersion();
```

Comment:
shouldn't we do this check before final result is done ? or is the intention to always show it - even for runs that reuse the cache?

Reply:
It's actually misnamed (historical reasons), renamed it to `informLatestVersion()`. The (async) check was already started in the beginning, that line is just retrieving the result.